### PR TITLE
#11342: Replace tt_lib  with ttnn function  in experimental/functional 

### PR DESCRIPTION
**Pull request by bharane-ab**
### Ticket
Link to Github Issue #11342 

### Problem description
Update functional files with ttnn support

### What's changed
 functional_mistral
 functional_roberta
 functional_unet
 functional_vit

### Checklist
- [x] Post commit CI passes
- [x] T 3000 model test
- [x] post commit model test
- [ ] single card demo test

### CODE REVIEW

File: models/experimental/functional_vit/tt/ttnn_optimized_vit_highres.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def vit_patch_embeddings(
     stride_w = 1
 
     pixel_values = ttnn.reshape(pixel_values, (batch_size, img_h, img_w // patch_size, 4 * patch_size))
-    pixel_values = ttl.tensor.fold(pixel_values, stride_h, stride_w)
+    pixel_values = ttnn.tensor.fold(pixel_values, stride_h, stride_w)
```
Comment (KalaivaniMCW) [inline models/experimental/functional_vit/tt/ttnn_optimized_vit_highres.py:37]: pls check this api in repo

File: models/experimental/functional_vit/tt/ttnn_functional_vit_highres.py [hunk 2]
```diff
@@ -26,7 +25,7 @@ def vit_patch_embeddings(config, pixel_values, *, parameters, unittest_check=Fal
     stride_w = 1
 
     pixel_values = ttnn.reshape(pixel_values, (batch_size, img_h, img_w // patch_size, 4 * patch_size))
-    pixel_values = ttl.tensor.fold(pixel_values, stride_h, stride_w)
+    pixel_values = ttnn.tensor.fold(pixel_values, stride_h, stride_w)
```
Comment (KalaivaniMCW) [inline models/experimental/functional_vit/tt/ttnn_functional_vit_highres.py:28]: check this

File: models/experimental/functional_mistral/tt/ttnn_functional_attention.py [hunk 3]
```diff
@@ -15,21 +14,21 @@ def apply_rotary_emb(xq, xk, bcast_freq_xq, bcast_freq_xk, device, mem_config):
...
-    xq = tt_lib.tensor.complex_tensor(xq_real, xq_img)
+    xq = ttnn.complex_tensor(xq_real, xq_img)
 
     xq_real.deallocate()
     xq_img.deallocate()
 
     xk_real = torch_to_tt_tensor_rm(t_xk[..., :, :, ::2], device)
     xk_img = torch_to_tt_tensor_rm(t_xk[..., :, :, 1::2], device)
-    xk = tt_lib.tensor.complex_tensor(xk_real, xk_img)
+    xk = tttnn.complex_tensor(xk_real, xk_img)
```
Comment (KalaivaniMCW) [inline models/experimental/functional_mistral/tt/ttnn_functional_attention.py:24]: spelling

File: models/experimental/functional_mistral/tt/ttnn_functional_attention.py [hunk 4]
```diff
@@ -15,21 +14,21 @@ def apply_rotary_emb(xq, xk, bcast_freq_xq, bcast_freq_xk, device, mem_config):
...
 
     xk_real.deallocate()
     xk_img.deallocate()
 
-    xq_out = tt_lib.tensor.complex_mul(xq, bcast_freq_xq, output_mem_config=mem_config)
+    xq_out = ttnn.mul(xq, bcast_freq_xq, output_mem_config=mem_config)
```
Comment (KalaivaniMCW) [inline models/experimental/functional_mistral/tt/ttnn_functional_attention.py:29]: memory_config= kw ???

File: models/experimental/functional_mistral/tt/ttnn_functional_attention.py [hunk 5]
```diff
@@ -15,21 +14,21 @@ def apply_rotary_emb(xq, xk, bcast_freq_xq, bcast_freq_xk, device, mem_config):
...
 
-    xk_out = tt_lib.tensor.complex_mul(xk, bcast_freq_xk, output_mem_config=mem_config)
+    xk_out = ttnn.mul(xk, bcast_freq_xk, output_mem_config=mem_config)
```
Comment (KalaivaniMCW) [inline models/experimental/functional_mistral/tt/ttnn_functional_attention.py:31]: memory_config= kw ???

Comment (bharane-ab) [reply models/experimental/functional_mistral/tt/ttnn_functional_attention.py:31]: till now there is no equivalent op for this mug in main

File: models/experimental/functional_unet/unet_utils.py [hunk 6]
```diff
@@ -5,9 +5,7 @@
 import torch
 import ttnn
 
-import tt_lib as ttl
-import tt_lib.fallback_ops
```
Comment (KalaivaniMCW) [inline models/experimental/functional_unet/unet_utils.py:9]: is this used in the file anywhere?

File: models/experimental/functional_unet/unet_utils.py [hunk 7]
```diff
@@ -5,9 +5,7 @@
...
-from tt_lib import profiler
+import ttnn.fallback_ops
```
Comment (KalaivaniMCW) [inline models/experimental/functional_unet/unet_utils.py:8]: don't think this is a valid import. remove it.

**Review by KalaivaniMCW — CHANGES_REQUESTED** [review]
Please check the diff and address comments. 
if op does not exist as ttnn. , the check in ttnn/deprecated and use ttnn.experimental

File: models/experimental/functional_unet/tt/unet_shallow_ttnn.py [hunk 8]
```diff
@@ -6,8 +6,7 @@
 import torch.nn as nn
 import ttnn
 
-import tt_lib.fallback_ops
-from tt_lib import profiler
+import ttnn.fallback_ops
```
Comment (umadevimcw) [inline models/experimental/functional_unet/tt/unet_shallow_ttnn.py:9]: Is this working?

File: models/experimental/functional_unet/tt/unet_shallow_torch.py [hunk 9]
```diff
@@ -6,9 +6,7 @@
 import torch.nn as nn
 import ttnn
 
-import tt_lib as ttl
-import tt_lib.fallback_ops
-from tt_lib import profiler
+import ttnn.fallback_ops
```
Comment (umadevimcw) [inline models/experimental/functional_unet/tt/unet_shallow_torch.py:9]: is ttnn.fallback_ops import working?

**Review by umadevimcw — CHANGES_REQUESTED** [review]

File: models/experimental/functional_vit/tt/ttnn_functional_vit.py [hunk 10]
```diff
@@ -27,7 +26,7 @@ def vit_patch_embeddings(config, pixel_values, *, parameters, unittest_check=Fal
     stride_w = 1
 
     pixel_values = ttnn.reshape(pixel_values, (batch_size, img_h, img_w // patch_size, 4 * patch_size))
-    pixel_values = ttl.tensor.fold(pixel_values, stride_h, stride_w)
+    pixel_values = ttnn.tensor.fold(pixel_values, stride_h, stride_w)
```
[the remark this answers is not in the document]
Comment (bharane-ab) [reply models/experimental/functional_vit/tt/ttnn_functional_vit.py]: I have changed it to experimental

File: models/experimental/functional_vit/tt/ttnn_functional_vit_highres.py (hunk 2 above)
Comment (bharane-ab) [reply models/experimental/functional_vit/tt/ttnn_functional_vit_highres.py]: I have changed it to experimental

File: models/experimental/functional_vit/tt/ttnn_optimized_vit_highres.py (hunk 1 above)
Comment (bharane-ab) [reply models/experimental/functional_vit/tt/ttnn_optimized_vit_highres.py]: I have changed it to experimental

File: models/experimental/functional_unet/tt/unet_shallow_ttnn.py (hunk 8 above)
Comment (bharane-ab) [reply models/experimental/functional_unet/tt/unet_shallow_ttnn.py]: no

File: models/experimental/functional_unet/demo/demo.py [hunk 11]
```diff
@@ -18,9 +18,7 @@
 from models.experimental.functional_unet.unet_utils import create_custom_preprocessor
 
 import time
-import tt_lib as ttl
 import os
-from tt_lib import profiler
```
Comment (KalaivaniMCW) [inline models/experimental/functional_unet/demo/demo.py:23]: Is this import being used? if so, keep this for now.

File: models/experimental/functional_vit/tt/ttnn_functional_vit.py [hunk 12]
```diff
@@ -27,7 +26,7 @@ def vit_patch_embeddings(config, pixel_values, *, parameters, unittest_check=Fal
     stride_w = 1
 
     pixel_values = ttnn.reshape(pixel_values, (batch_size, img_h, img_w // patch_size, 4 * patch_size))
-    pixel_values = ttl.tensor.fold(pixel_values, stride_h, stride_w)
+    pixel_values = ttnn.experimental.tensor.fold(pixel_values, stride_h, stride_w)
```
Comment (KalaivaniMCW) [inline models/experimental/functional_vit/tt/ttnn_functional_vit.py:29]: I think this op has changed https://github.com/tenstorrent/tt-metal/commit/7c67e3ec6de8bf9fbe1d0f218e6013db59bf9ea9

File: models/experimental/functional_vit/tt/ttnn_functional_vit_highres.py [hunk 13]
```diff
@@ -26,7 +25,7 @@ def vit_patch_embeddings(config, pixel_values, *, parameters, unittest_check=Fal
     stride_w = 1
 
     pixel_values = ttnn.reshape(pixel_values, (batch_size, img_h, img_w // patch_size, 4 * patch_size))
-    pixel_values = ttl.tensor.fold(pixel_values, stride_h, stride_w)
+    pixel_values = ttnn.experimental.tensor.fold(pixel_values, stride_h, stride_w)
```
Comment (KalaivaniMCW) [inline models/experimental/functional_vit/tt/ttnn_functional_vit_highres.py:28]: I think this op has changed https://github.com/tenstorrent/tt-metal/commit/7c67e3ec6de8bf9fbe1d0f218e6013db59bf9ea9

File: models/experimental/functional_unet/demo/demo.py (hunk 11 above)
Comment (bharane-ab) [reply models/experimental/functional_unet/demo/demo.py:23]: as it is not used in the file I have removed it

**Review by KalaivaniMCW — APPROVED** [review]
Approved

**Review by umadevimcw — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]

**Review by VirdhatchaniKN — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

File: models/experimental/functional_unet/tt/unet_shallow_ttnn.py [hunk 14]
```diff
@@ -4,8 +4,6 @@
 
 import ttnn
 
-from tt_lib import profiler
```
Comment (esmalTT) [inline models/experimental/functional_unet/tt/unet_shallow_ttnn.py:7]: There is code using `profiler` below

**Review by esmalTT — CHANGES_REQUESTED** [review]
Thanks for the updates. Once the comments are addressed, please run nightly pipeline confirming everything is OK.

**Review by esmalTT — APPROVED** [review]

**Review by ayerofieiev-tt — APPROVED** [review]

**Pull request merged**